apps/mobile/src/features/send/components/asset-picker/asset-picker.tsx

```diff
@@ -1,45 +1,120 @@
+import { ReactNode, useMemo } from 'react';
+
+import { renderAsset } from '@/features/balances/assets/render-assets';
 import {
   BitcoinBalance,
   BitcoinBalanceByAccount,
 } from '@/features/balances/bitcoin/bitcoin-balance';
 import { StacksBalance, StacksBalanceByAccount } from '@/features/balances/stacks/stacks-balance';
+import { sortSip10Balances } from '@/features/balances/utils/sort-sip10-balances';
 import { AssetPickerItem } from '@/features/send/components/asset-picker/asset-picker-item';
-import { SendableAsset } from '@/features/send/types';
+import {
+  useSip10AccountBalance,
+  useSip10TotalBalance,
+} from '@/queries/balance/sip10-balance.query';
 import { Account } from '@/store/accounts/accounts';
+import { BottomSheetFlashList } from '@gorhom/bottom-sheet';
+
+import { btcAsset, stxAsset } from '@leather.io/constants';
+import { FungibleCryptoAsset } from '@leather.io/models';
+import { Sip10Balance } from '@leather.io/services';
 
 interface AssetListProps {
   account: Account | null;
-  onSelectAsset(asset: SendableAsset, assetElementOffsetTop: number | null): void;
+  onSelectAsset(asset: FungibleCryptoAsset, assetElementOffsetTop: number | null): void;
+}
+
+interface AssetPickerFlashListProps {
+  sip10Data: ReturnType<typeof useSip10TotalBalance>;
+  header: ReactNode;
+  handleSelectAsset(asset: FungibleCryptoAsset): (top: number | null) => void;
+}
+
+function AssetPickerFlashList({ sip10Data, header, handleSelectAsset }: AssetPickerFlashListProps) {
+  const sip10Memo = useMemo(() => {
+    if (sip10Data.state === 'success') return sip10Data.value.sip10s.sort(sortSip10Balances);
+    return [];
+  }, [sip10Data]);
+
+  return (
+    <BottomSheetFlashList<Sip10Balance>
```

Comment:
It would be good to handle loading / error here using `ListEmptyComponent`  / the `state.isLoading`